test/javascripts/integration/components/upcoming-events-list-test.gjs

```diff
@@ -163,6 +163,41 @@ module("Integration | Component | upcoming-events-list", function (hooks) {
     );
   });
 
+  test("Uses custom category name from 'map_events_title'", async function (assert) {
+    pretender.get("/discourse-post-event/events", () => {
+      return response({ events: [] });
+    });
+
+    this.siteSettings.map_events_title =
+      '[{"category_slug": "announcements", "custom_title": "Upcoming Announcements"}]';
+
+    await render(<template><UpcomingEventsList /></template>);
+    this.appEvents.trigger("page:changed", { url: "/c/announcements" });
+
+    assert.strictEqual(
+      query(".upcoming-events-list__heading").innerText,
+      "Upcoming Announcements",
+      "it sets 'Upcoming Announcements' as the title in 'c/announcements'"
+    );
```

Comment:
qunit-dom assertions are the only recommended way to test text/html/existence/etc going forward

```suggestion
    assert
      .dom(".upcoming-events-list__heading")
      .hasText(
        "Upcoming Announcements",
        "sets 'Upcoming Announcements' as the title in 'c/announcements'"
      );
```

Reply:
good to know, thank you!